featbit/featbit
# ✨feat: readiness endpoint for the evaluation server

# Adds a readiness endpoint to the evaluation server
The endpoint will check:
- that MongoDB is available
- that Redis is available
- that Kafka is available (if running FeatBit pro)

*If any of these services are unavailable, it will return a 503.

## Why this is necessary

This endpoint will indicate that the evaluation server's dependencies are available and it is ready to process requests, not just that the evaluation server is running. This is useful for having a Kubernetes readiness probe (https://spacelift.io/blog/kubernetes-readiness-probe).

## How to test

1. Start the docker containers for FeatBit (redis, Mongodb, kafka)
2. Start the evaluation server project
3. Curl http://localhost:5100/health/readiness. You should get a 200 status code.

### How to test MongoDB being down

1. Stop the MongoDB container
3. Curl http://localhost:5100/health/readiness. You should get a 503 status code.

### How to test Redis being down

1. Stop the Redis container
3. Curl http://localhost:5100/health/readiness. You should get a 503 status code.

### How to test Kafka being down

1. Set `IS_PRO` to `true`
2. Restart the evaluation server project
3. Stop the Kafka container
4. Curl http://localhost:5100/health/readiness. You should get a 503 status code.
